### PR TITLE
API-5762 Payload signer implementation

### PR DESCRIPTION
**Pull request by apeirats-va**
Adds a `PayloadSigner` component that signs a VC payload with a configurable private key, and generates a JWS response.

The JWS payload can be optionally compressed (using Java's ZLIB implementation).

API-5763 will register this as a Spring `Component`  and integrate it with `PatientController`.

### CODE REVIEW

File: smart-cards-tests/pom.xml [hunk 1]
```diff
@@ -106,6 +106,8 @@
                 <argument>-Dlogging.level.root=INFO</argument>
                 <argument>-Ddata-query.internal-url=http://localhost:8777</argument>
                 <argument>-Ddata-query.r4-base-path=/</argument>
+                <argument>-Djwk-set.private-json='{\"keys\":[{\"kty\":\"EC\",\"d\":\"5h-A-oFDpePWC9zZ54_sWANZlwDxppu2-DQuAuvW3Bw\",\"use\":\"sig\",\"crv\":\"P-256\",\"kid\":\"gSb6VludekqwZS_0osHNxAK2y2Rua1V_aGWNFaTvgNM\",\"x\":\"WcBuVJRGiCvhEdU6nbKysIxEjQOqzaH-kiIZoIpFNfE\",\"y\":\"FiWqBp6ilDBJ-VEnZsgHTK4fYKmB841n2wNMfvFZB2E\",\"alg\":\"ES256\"}]}'</argument>
```
Comment (jblefkove-va) [inline smart-cards-tests/pom.xml:109]: are the enclosing single-quotes necessary here?

File: smart-cards/src/main/java/gov/va/api/health/smartcards/Compressors.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  /** Compress a payload with Java's ZLIB implementation. */
+  public static byte[] deflate(byte[] input) {
+    checkNotNull(input);
+    Deflater deflater = new Deflater();
+    deflater.setInput(input);
+    deflater.finish();
+
+    byte[] buffer = new byte[BUFFER_SIZE];
+    ByteArrayOutputStream bos = new ByteArrayOutputStream();
```
Comment (jblefkove-va) [inline smart-cards/src/main/java/gov/va/api/health/smartcards/Compressors.java:23]: even though closing a `ByteArrayOutputStream` has no effect, I feel like this should still be declared via try-with-resources

File: smart-cards/src/main/java/gov/va/api/health/smartcards/Compressors.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  /** Decompress a payload with Java's ZLIB implementation. */
+  @SneakyThrows
+  public static byte[] inflate(byte[] input) {
+    checkNotNull(input);
+    Inflater inflater = new Inflater();
+    inflater.setInput(input);
+    byte[] buffer = new byte[BUFFER_SIZE];
+    ByteArrayOutputStream bos = new ByteArrayOutputStream();
```
Comment (jblefkove-va) [inline smart-cards/src/main/java/gov/va/api/health/smartcards/Compressors.java:42]: ditto above comment

File: smart-cards/src/main/java/gov/va/api/health/smartcards/JwksProperties.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+  public JWK currentPrivateJwk() {
+    return jwksPrivate.getKeyByKeyId(currentKeyId);
+  }
+
+  public JWK currentPublicJwk() {
+    return jwksPublic.getKeyByKeyId(currentKeyId);
+  }
+
+  public JWKSet jwksPublic() {
```
Comment (jblefkove-va) [inline smart-cards/src/main/java/gov/va/api/health/smartcards/JwksProperties.java:54]: won't lombok generate this one?

File: smart-cards/src/main/java/gov/va/api/health/smartcards/vc/PayloadClaimsWrapper.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor
+@JsonAutoDetect(
+    fieldVisibility = JsonAutoDetect.Visibility.ANY,
+    isGetterVisibility = JsonAutoDetect.Visibility.NONE)
+@Schema
```
Comment (jblefkove-va) [inline smart-cards/src/main/java/gov/va/api/health/smartcards/vc/PayloadClaimsWrapper.java:19]: schema tag is unnecessary. (Pretty sure the `@JsonAutoDetect` is mainly used for swagger generation as well.) `@AllArgsConstructor` can be private for good measure.

File: smart-cards/src/main/java/gov/va/api/health/smartcards/JwksProperties.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      @Value("${jwk-set.current-key-id}") String currentKeyId) {
+    checkState(!"unset".equals(jwksPrivateJson), "jwk-set.private-json is unset");
+    checkState(!"unset".equals(currentKeyId), "jwk-set.current-key-id is unset");
+    this.currentKeyId = currentKeyId;
+    jwksPrivate = JWKSet.parse(cleanupJson(jwksPrivateJson));
+    jwksPublic = jwksPrivate.toPublicJWKSet();
+  }
+
+  /** Deal with quotes for json provided through CLI properties. */
+  private static String cleanupJson(String json) {
```
Comment (jblefkove-va) [inline smart-cards/src/main/java/gov/va/api/health/smartcards/JwksProperties.java:37]: this seems suspicous

File: smart-cards-tests/pom.xml (hunk 1 above)
Comment (apeirats-va) [reply smart-cards-tests/pom.xml:109]: On mingw I did not need them. Jenkins' build was failing as Linux/bash messed with the quotes, so the json parser failed. This wrapping with single quotes, along with the `cleanupJson` method, solved that problem. I'm unsure if this is the correct solution, though.

File: smart-cards/src/main/java/gov/va/api/health/smartcards/JwksProperties.java (hunk 4 above)
Comment (apeirats-va) [reply smart-cards/src/main/java/gov/va/api/health/smartcards/JwksProperties.java:54]: It does. Refactor gone wrong made this redundant.

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**